File: 실버 3 (5문제)/9095_1, 2, 3 더하기.c
// 참고 링크 : https://moon-gd.tistory.com/52
#define _CRT_SECURE_NO_WARNINGS
#include <stdio.h>
#define MAX 11

int cases[MAX]= {0, 1, 2, 4};

int main() {
	for (int i = 4; i < MAX; i++) {
		cases[i] = cases[i - 1] + cases[i - 2] + cases[i - 3];
	}

	int test_num = 0, temp;
	scanf("%d", &test_num);

	for (int i = 0; i < test_num; i++) {
		scanf("%d", &temp);
		printf("%d\n", cases[temp]);
	}

	return 0;
}

File: 실버 3 (5문제)/9095_1, 2, 3 더하기.py
# 참고 링크 : https://moon-gd.tistory.com/52
count = [0] * 11

count[1] = 1
count[2] = 2
count[3] = 4
for i in range(4, 11):
    count[i] = count[i - 1] + count[i - 2] + count[i - 3]

import sys
q_input = sys.stdin.readline

test_num = int(q_input())

for _ in range(test_num):
    print(count[int(q_input())])
